blues/reporters.py

```diff
@@ -536,3 +354,99 @@ def report(self, simulation, state):
             self._out.flush()
         except AttributeError:
             pass
+
+class NetCDF4Reporter(parmed.openmm.reporters.NetCDFReporter):
```

Comment:
For my own understanding, could you elaborate on the use case differences between NetCDF and HDF5? For example, when would you use one over another?